packages/flutter/test/painting/image_cache_test.dart

```diff
@@ -169,7 +170,15 @@ void main() {
         return completer1;
       }) as TestImageStreamCompleter;
 
-      imageCache.clear();
+      final FlutterImageCache flutterImageCache = imageCache as FlutterImageCache;
+      expect(flutterImageCache.locationForKey(testImage).pending, true);
+      expect(flutterImageCache.locationForKey(testImage).weak, true);
+      flutterImageCache.clear();
+      expect(flutterImageCache.locationForKey(testImage).pending, false);
+      expect(flutterImageCache.locationForKey(testImage).weak, true);
+      flutterImageCache.clearWeakImages();
+      expect(flutterImageCache.locationForKey(testImage).pending, false);
+      expect(flutterImageCache.locationForKey(testImage).weak, false);
```

Comment:
clear does remove the pending image, but this test no longer captures how to check if an image is pending or not.

This test could be independently rewritten without these changes to use the existing `containsKey` API, although the new API offers more robust ways to check if an image is pending.